### PR TITLE
Add note that Spring support is unavailable in 8.6+

### PR DESCRIPTION
**Pull request by akeller**
## Description

Spring support with Zeebe Process Test is not available from 8.6+ and comes from a community-supported SDK. This is not clear and causing support tickets, so we add a note that this is unavailable.

In addition, I removed the step linking to the fully supported Spring Zeebe SDK. It's not applicable and redundant (you get this info on the ZPT repo README, including the annotation `spring-zeebe-test`.). 

More internal context can be found [here](https://camunda.slack.com/archives/C06VC96GMNE/p1734021018422249?thread_ts=1729598950.187899&cid=C06VC96GMNE).

Related to https://github.com/camunda/zeebe-process-test/pull/1282.

## When should this change go live?

- [x] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [x] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

File: docs/components/best-practices/development/testing-process-definitions.md [hunk 1]
```diff
@@ -56,15 +56,19 @@ When using Java, most customers use Spring Boot. While this is a common setup fo
 
 ### Technical setup using Spring
 
+:::caution
+Spring support with Zeebe Process Test uses the community-supported Spring Zeebe SDK.
+Spring support is not available for Camunda 8.6+.
```
Comment (saig0) [inline docs/components/best-practices/development/testing-process-definitions.md:61]: :wrench: We could add more details. 

```suggestion
Spring support with Zeebe Process Test uses the community-maintained project Spring Zeebe.
The new Camunda Spring SDK (Camunda 8.6+) is not supported. You could still use the testing library but without hooking into the Spring lifecycle.
```

**Review by saig0 — APPROVED** [review]
@akeller looks good. :+1: 

I added a suggestion. Please have a look.

File: docs/components/best-practices/development/testing-process-definitions.md (hunk 1 above)
Comment (akeller) [reply docs/components/best-practices/development/testing-process-definitions.md:61]: This is good clarity, thank you @saig0!

**Review by saig0 — CHANGES_REQUESTED** [review]

File: versioned_docs/version-8.6/components/best-practices/development/testing-process-definitions.md [hunk 2]
```diff
@@ -56,15 +56,19 @@ When using Java, most customers use Spring Boot. While this is a common setup fo
 
 ### Technical setup using Spring
 
+:::caution
+Spring support with Zeebe Process Test uses the community-supported Spring Zeebe SDK.
+Spring support is not available for Camunda 8.6+.
```
[the remark this answers is not in the document]
Comment (akeller) [reply versioned_docs/version-8.6/components/best-practices/development/testing-process-definitions.md:61]: Thank you, moving too fast this morning and thought this was the readme PR 😰

**Review by saig0 — APPROVED** [review]
:rocket:

**Pull request merged**